controllers/controller_shared.go

```diff
@@ -87,9 +87,11 @@ func GetScopedMatchingInstances(ctx context.Context, k8sClient client.Client, cr
 		if !selected {
 			continue
 		}
+
+		doReadyCheck := instance.Annotations["grafana/skip-ready-check"] == ""
```

Comment:
Let's add the comment explaining usage of readiness checks here.
Also, let's rename it from readyCheck to readinessCheck.